### ISSUE TITLE
TypeError: params.hasOwnProperty is not a function

### ISSUE THREAD

**Issue by jtojnar** [havard/node-openid#148]
After upgrading to node 6.0.0, `openid` no longer works.

```
/home/jtojnar/projects/pengu/node_modules/openid/openid.js:1233
    if(params.hasOwnProperty(key) && key != 'openid.mode')
              ^

TypeError: params.hasOwnProperty is not a function
    at _checkSignatureUsingProvider (/home/jtojnar/projects/pengu/node_modules/openid/openid.js:1233:15)
    at _checkSignature (/home/jtojnar/projects/pengu/node_modules/openid/openid.js:1168:5)
    at _verifyAssertionAgainstProviders (/home/jtojnar/projects/pengu/node_modules/openid/openid.js:1132:12)
    at /home/jtojnar/projects/pengu/node_modules/openid/openid.js:1097:7
    at /home/jtojnar/projects/pengu/node_modules/openid/openid.js:546:7
    at /home/jtojnar/projects/pengu/node_modules/openid/openid.js:412:16
    at Request._callback (/home/jtojnar/projects/pengu/node_modules/openid/openid.js:175:7)
    at Request.self.callback (/home/jtojnar/projects/pengu/node_modules/request/request.js:200:22)
    at emitTwo (events.js:106:13)
    at Request.emit (events.js:191:7)
```

The `params` variable originates on [line 908](https://github.com/havard/node-openid/blob/419b1c6b049758208350ab4495824ece180a857f/openid.js#L908). It looks like querystring parser used by `url.parse` now uses [a blank object](https://github.com/nodejs/node/commit/dba245f79622c0c5877825f7cd11667be3548e8e).

**Comment by artemio** [comment]
nodejs/node#6055

**Comment by mscdex** [comment]
PR for fix in #149.
